vendors/swedbank-pay-sdk-php/src/SwedbankPay/Framework/AbstractSimpleDataObject.php

```diff
@@ -0,0 +1,205 @@
+     * Assigns Data value to the specified data
+     *
+     * @param string The data key to assign the value to
+     * @param mixed  The value to set
+     * @access public
+     * @return $this
+     */
+    public function __set($key, $value)
+    {
+        $this->data[$key] = $value;
+        return $this;
+    }
+
+    /**
+     * Whether or not an data exists by key
+     *
+     * @param string An data key to check for
+     * @access public
+     * @return boolean
+     * @abstracting ArrayAccess
+     */
+    public function __isset($key)
+    {
+        return isset($this->data[$key]);
+    }
+
+    /**
+     * Unsets an data by key
+     *
+     * @param string The key to unset
+     * @access public
+     * @return $this
+     */
+    public function __unset($key)
+    {
+        unset($this->data[$key]);
+        return $this;
+    }
+
+    /**
+     * Assigns Data value to the specified offset
+     *
+     * @param string The offset to assign the value to
+     * @param mixed  The value to set
+     * @access public
+     * @abstracting ArrayAccess
+     * @return $this
+     */
+    public function offsetSet($offset, $value)
+    {
+        if (is_null($offset)) {
+            $this->data[] = $value;
+            return $this;
+        }
+
+        $this->data[$offset] = $value;
+        return $this;
+    }
+
+    /**
+     * Whether or not an offset exists
+     *
+     * @param string An offset to check for
+     * @access public
+     * @return boolean
+     * @abstracting ArrayAccess
+     */
+    public function offsetExists($offset)
+    {
+        return isset($this->data[$offset]);
+    }
+
+    /**
+     * Unsets an offset
+     *
+     * @param string The offset to unset
+     * @access public
+     * @abstracting ArrayAccess
+     * @return $this
+     */
+    public function offsetUnset($offset)
+    {
+        if ($this->offsetExists($offset)) {
+            unset($this->data[$offset]);
+        }
+        return $this;
+    }
+
+    /**
+     * Returns the value at specified offset
+     *
+     * @param string The offset to retrieve
+     * @access public
+     * @abstracting ArrayAccess
+     * @return mixed
+     */
+    public function offsetGet($offset)
+    {
+        return $this->offsetExists($offset) ? $this->data[$offset] : null;
+    }
+
+    /**
+     * Return Data Object data in array format.
+     *
+     * @SuppressWarnings(PHPMD.CamelCaseMethodName)
+     *
+     * @param DataTransferObjectInterface|array|null $data
+     * @return array
+     */
+    public function __toArray($data = null)
+    {
+        if (is_null($data)) {
+            $data = $this->data;
+        }
+
+        $hasToArray = function ($model) {
+            return is_object($model) && method_exists($model, '__toArray') && is_callable([$model, '__toArray']);
+        };
+
+        foreach ($data as $key => $value) {
+            if ($hasToArray($value)) {
+                $data[$key] = call_user_func([$value, '__toArray']);
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [The use of function call_user_func() is discouraged](https://app.codacy.com/gh/SwedbankPay/swedbank-pay-woocommerce-checkout/pullRequest?prid=5510424)